src/session.rs

```diff
@@ -29,7 +29,28 @@ impl Session {
             })
             .await
             .expect("Driver should not panic")?;
-        return Ok(RequestResult { inner: result });
+        Ok(RequestResult { inner: result })
+    }
+
+    async fn execute_with_values(
+        &self,
+        request: Py<PyString>,
+        values: Py<PyAny>,
+    ) -> PyResult<RequestResult> {
+        let request_string = Python::with_gil(|py| request.to_str(py))?.to_string();
+        let session_clone = Arc::clone(&self._inner);
+        let result = RUNTIME
+            .spawn(async move {
+                session_clone
+                    .query_unpaged(request_string, PyAnyWrapper(values))
+                    .await
+                    .map_err(|e| {
+                        PyRuntimeError::new_err(format!("Failed to deserialize metadata: {}", e))
+                    })
+            })
+            .await
+            .expect("Driver should not panic")?;
```

Comment:
🔧 It makes little sense to spawn a task on a runtime only to await it straightaway. I'd suggest awaiting the query future directly, without using `RUNTIME`.